docs/pages/access-controls/idps/saml-attribute-mapping.mdx

```diff
@@ -0,0 +1,194 @@
+---
+
+Attribute mapping configures Teleport SAML Identity Provider to assert custom user attributes in SAML response
+The Teleport SAML IdP supports three configurable fields for attribute mapping:
+- `name`: Name of the outgoing attribute. Required. Name should be unique across attribute mapping.
+- `value`: Value defined using a predicate expression, which can reference Teleport usernames, roles or traits. Required.
+- `name_format`: SAML attribute name format. Optional. The following formats are supported:
+    - `unspecified`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified`. Used as a default value.
+    - `uri`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:uri`.
+    - `basic`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:basic`.
+
+The values can be configured with `saml_idp_service_provider` spec.
+```
+kind: saml_idp_service_provider
+metadata:
+  name: example.com
+spec:
+  entity_id: https://example.com/saml/metadata
+  acs_url: https://example.com/saml/metadata
+  attribute_mapping:
+  - name: username
+    value: uid
+  - name: firstname
+    name_format: basic # optional, unspecified used as default if no value is provided.
+    value: user.spec.traits.firstname
+  - name: groups
+    name_format: urn:oasis:names:tc:SAML:2.0:attrname-format:basic # optional, full urn format.
+    value: user.spec.roles
+```
+
+## Attribute value evaluation
+
+Attribute values are authored using predicate expressions. During an SSO request, the expressions will be parsed from the
+Service Provider metadata and passed to predicate evaluator for evaluation with the authenticated user context.
+The resulting values are asserted in the SAML response under the requested attribute name.
+
+Given a correct and supported predicate expression, attributes will be mapped as long as the requested attributes are present in Teleport.
+Attribute mapping that points to a non-existent value will not be included in SAML assertion.
+
+## Evaluation context
+
+The following user attributes are available for mapping between Teleport IdP and Service Providers:
+
+| Attributes | Description                   | Syntax                                                       |
+| ---------- | ------------------------------| ------------------------------------------------------------ |
+| Username   | User name.                    | `uid` or `user.metadata.name`.                               |
+| Roles      | User roles.                   | `eduPersonAffiliation` or `user.spec.roles`.                 |
+| Traits     | All values under user traits. | `user.spec.traits.firstname`, `user.spec.traits.groups` etc. |
+
+
+## Predicate expressions
+
+Predicate expressions for attribute mapping are evaluated against user attributes that can be accessed using evaluation context listed above.
+
+Expression syntax are similar to how expressions are used in [Login Rules](../login-rules/reference.mdx). But not all of the
```

Comment:
Nit: I'm not sure we should mention login rules here tbh. This assumes readers have familiarity with login rules which may not be the case. I would just keep the sentence "The supported functions ...".